src/main/java/org/spongepowered/common/data/key/KeyRegistry.java

```diff
@@ -294,6 +295,8 @@ private static void generateKeyMap() {
         keyMap.put("hide_can_destroy", makeSingleKey(Boolean.class, Value.class, of("HideCanDestroy")));
         keyMap.put("hide_can_place", makeSingleKey(Boolean.class, Value.class, of("HideCanPlace")));
         keyMap.put("hide_miscellaneous", makeSingleKey(Boolean.class, Value.class, of("HideMiscellaneous")));
+        keyMap.put("beacon_primary_effect", makeSingleKey(PotionEffectType.class, OptionalValue.class, of("BeaconPrimaryEffect")));
+        keyMap.put("beacon_secondary_effect", makeSingleKey(PotionEffectType.class, OptionalValue.class, of("BeaconSecondaryEffect")));
```

Comment:
The classes used here are of the incorrect type. Generically, you'd have to use:
```
keyMap.put("beacon_primary_effect", makeSingleKey(Optional.class, OptionalValue.class, of("BeaconPrimaryEffect"));
```
Since the element class type isn't `PotionEffectType` but rather `Optional`